torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -231,16 +234,22 @@ def na_forward(self, feat: TensorData) -> TensorData:
                     feat.device)
             else:
                 raise ValueError(f"Unsupported NA strategy {self.na_strategy}")
-            if isinstance(feat, _MultiTensor):
+            fill_values.append(fill_value)
+
+        if isinstance(feat, _MultiTensor):
+            for col, fill_value in enumerate(fill_values):
                 feat.fillna_col(col, fill_value)
-            else:
-                column_data = feat[:, col]
-                na_mask = get_na_mask(column_data)
-                if na_mask.ndim == 2:
-                    na_mask = na_mask.any(dim=-1)
-                assert na_mask.ndim == 1
-                assert len(na_mask) == len(column_data)
-                column_data[na_mask] = fill_value
+        else:
+            if na_mask.ndim == 3:
+                # when feat is 3D, it is faster to iterate over columns
+                for col, fill_value in enumerate(fill_values):
+                    col_data = feat[:, col]
+                    col_na_mask = na_mask[:, col].any(dim=-1)
+                    col_data[col_na_mask] = fill_value
+            else:  # na_mask.ndim == 2
+                fill_values = torch.tensor(fill_values, device=feat.device)
```

Comment:
`torch.stack(fill_values)`?

Reply:
`torch.stack` would be correct for `na_mask.ndim == 3`. Here it would generate an error, because `fill_values` is a list of ints/floats, not a list of tensors. 😉

Reply:
Ah I see. Thanks for clarifying